### ISSUE TITLE
Insufficient Permission error when using a gcs PipelineResource

### ISSUE THREAD

**Issue by dustym** [tektoncd/pipeline#1748]
# Expected Behavior

We have a PipelineRun that outputs to a storage resource of type `gcs`. We authenticate this PipelineResource with a secret in the spec that looks like this:

```
  secrets:
    - fieldName: GOOGLE_APPLICATION_CREDENTIALS
      secretName: bucket-sa
      secretKey: ci-service.json
```

This has worked since we rolled out v0.7.0. We upgraded from v0.8.0 to v0.9.0 expecting this behavior to continue to work.

# Actual Behavior

After upgrading to v0.9.0, our PipelineRun started to fail when uploading this resource. I spent a while double checking our configuration. If I roll back to v0.8.0 everything would work again.

The error we see in the logs from the upload step is: 

```
ResumableUploadAbortException: 403 Insufficient Permission
```

# Steps to Reproduce the Problem

1. Create a PipelineResource that utilizes a GOOGLE_APPLICATION_CREDENTIALS secret to upload to a GCS bucket and try to upload to it

# Additional Info

I created a test TaskRun with the PipelineResource and made a previous step `sleep` indefinitely. I then connected into the  upload step. There is an environment variable for the secret specified above:

```
GOOGLE_APPLICATION_CREDENTIALS=/var/secret/bucket-sa/ci-service.json
```

However when I run `gcloud auth list` the service account in that key is not listed as available, only the k8s service account the pod is configured to use is available. A `gsutil cp` to the bucket test confirms insufficient permissions. If I run

```
gsutil auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
```

The test `gsutil cp` will then work as expected.

After googling a bit, I found GoogleCloudPlatform/cloud-sdk-docker#152 which is later referenced by tektoncd/pipeline#402 that was resolved by 2021783dd958054fe9cf5cc2584d5b731b9c1e09. 

The code for that change was removed in #1504 and it seems like there was no replacement for the `activate-service-account` behavior put back in place.

**Comment by sbwsg** [comment]
fyi @ImJasonH 

Looks like the gsutil wrapper we had was doing some heavy lifting around the GOOGLE_APPLICATION_CREDENTIALS env variable.

Possible next steps for solving this:
- Roll back #1504 
- Find a fix that uses the official gsutil image but that supports GOOGLE_APPLICATION_CREDENTIALS env var

**Comment by ImJasonH** [comment]
This is an unfortunate breakage, but I think the correct way forward is to depend on gcloud behavior and not paper over it with our own image behavior.

This means users should call `gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS` themselves (possibly as part of the containers run by the GCS resource) to make sure creds are available.

**Comment by dustym** [comment]
> This means users should call gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS themselves (possibly as part of the containers run by the GCS resource) to make sure creds are available.

@ImJasonH Does this mean I need to roll my own image for this to work according to documentation? If so is there some easy way to specify the default `GsutilImage` image (other than modifying the `tekton-pipelines-controller` `args`?

**Comment by ImJasonH** [comment]
For now you can set the gsutil image flag to what it was before. That's obviously not a valid long term solution.

For that, we should make the GCS resource call the necessary gcloud command to activate the SA specified in the env var, then invoke `google/cloud-sdk` as it does today.

**Comment by dustym** [comment]
Thanks for the fix @ImJasonH

**Comment by ImJasonH** [comment]
@dustym please reopen if this doesn't fix your issue

**Comment by mgreau** [comment]
We have upgraded our cluster from 0.8.0 to 0.9.2 and we do have the exact same issue:

```
$ kubectl logs stage-pages-pod-80e41e -c step-fetch-release-bucket-86gw7
Building synchronization state...
Caught non-retryable exception while listing gs://my-release-bucket./: AccessDeniedException: 403 Insufficient OAuth2 scope to perform this operation.
Acceptable scopes: https://www.googleapis.com/auth/cloud-platform
CommandException: Caught non-retryable exception - aborting rsync
```

I'm going to try https://github.com/tektoncd/pipeline/pull/1757 to see if that fixes our issue.

@ImJasonH @bobcatfish If #1757 is confirmed to be a fix, do you think it would be possible to do a 0.9.3 release?

Thanks

**Comment by dustym** [comment]
@mgreau Yeah we've been holding off on upgrading from 0.8.0 until the next release.

**Comment by mgreau** [comment]
I tried in several ways but I can't make it work.

I forked the repo and from https://github.com/tektoncd/pipeline/tree/v0.9.2 I cherry-picked https://github.com/tektoncd/pipeline/commit/cf8705b8872c4c491f6b44db2fb9ee213b21edae 
Then I followed the instructions and built the https://github.com/mgreau/pipeline/blob/v0.9.2-plus-fix-1757/release-v0.9.3-mgreau.yaml (where only the `gcs-fetcher` image is updated)

Then, I tried with the release-0.9.2.yaml file and by updating the `-gsutil-image` tag from `google/cloud-sdk` to `gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/gsutil:latest@sha256:0130ec562b897c5929123d4e14cd3271cd58102f1f411f52cb6f415088bf5944`, this is the https://github.com/mgreau/pipeline/blob/v0.9.2-plus-fix-1757/release-v0.9.2-gsutil-tekton.yaml

In both cases, I have the following error when applying the `TaskRun` yaml file:

```
$ kubectl get taskruns                                                                                                                                                                    130 ↵
NAME                          SUCCEEDED   REASON           STARTTIME   COMPLETIONTIME
stage-download-pages-093   False       CouldntGetTask   19m
```
```
status:
  conditions:
  - lastTransitionTime: "2020-01-16T20:17:42Z"
    message: 'Missing or invalid Task release-staging-env/download-page-publisher:
      translating Build to Pod: Error getting image manifest: UNAUTHORIZED: authentication
      required; [map[Action:pull Class: Name:infra/download-page-publisher Type:repository]]'
    reason: CouldntGetTask
    status: "False"
    type: Succeeded
  podName: ""
  startTime: "2020-01-16T20:17:42Z"
```

**Comment by mgreau** [comment]
To close the discussion here, I can confirm this issue is fixed in Tekton v0.10.0.

Thanks
